src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -129,13 +119,13 @@ void OTARequestor::OnQueryImageResponse(void * context, const QueryImageResponse
 
         if (err != CHIP_NO_ERROR)
         {
-            requestorCore->mOtaRequestorDriver->HandleError(UpdateFailureState::kQuerying, err);
+            requestorCore->RecordErrorUpdateState(UpdateFailureState::kQuerying, err);
```

Comment:
Added a utility `RecordErrorUpdateState` which would call `HandleError`, log state transitions as well as update server attributes. Anywhere we used to call `HandleError` will now be replaced by this function.